extension/content/opt-out-ext.js

```diff
@@ -1,104 +1,102 @@
-        }
-        if (xhr.status === 200) {
-            console.log("Response received as ", JSON.parse(xhr.response).predictions[0]);
-            if (JSON.parse(xhr.response).predictions[0]){
-                node.classList.add("processed-true");
-                let tweetText = node.querySelector(selector + ' > div ~ div > div ~ div');
-                styleTweet(tweetText, option);
-            }
-            else {
-                node.classList.add("processed-false");
-            }
-        } else {
-            console.log("Failed response", xhr);
-        }
-    };
-    xhr.send(
-        JSON.stringify({
-            texts: [node.innerText]
-        })
-    );
+const styleTweet = function (element, selectedOption) {
+  if (selectedOption.includes('tw')) element.classList.add('opt-out-tw');
+  else element.classList.remove('opt-out-tw');
+  if (selectedOption.includes('tc')) element.classList.add('opt-out-tc');
+  else element.classList.remove('opt-out-tc');
+  if (selectedOption.includes('tr')) element.classList.add('opt-out-trem');
+  else element.classList.remove('opt-out-trem');
 };
 
 /*
-Depending on `option` sets classes to tweet nodes
+function which calls server for given node, and depending on the response,
+applies pre-defined action
  */
-let option = "tw";
-styleTweet = function(element, option) {
-    if (option.includes('tw'))
-        element.classList.add("opt-out-tw");
-    else
-        element.classList.remove("opt-out-tw");
-    if (option.includes('tc'))
-        element.classList.add("opt-out-tc");
-    else
-        element.classList.remove("opt-out-tc");
-    if (option.includes('tr'))
-        element.classList.add("opt-out-trem");
-    else
-        element.classList.remove("opt-out-trem");
+const checkText = function (node) {
+  console.log('Sending Request');
+  const link = 'https://api.optoutools.com/predict';
+  const xhr = new XMLHttpRequest();
+  xhr.open('POST', link, true);
+  xhr.setRequestHeader('Content-type', 'application/json;charset=UTF-8');
+  xhr.withCredentials = true;
+  xhr.onreadystatechange = function (e) {
+    if (xhr.readyState !== 4) {
+      return;
+    }
+    if (xhr.status === 200) {
+      console.log(
+        'Response received as ',
+        JSON.parse(xhr.response).predictions[0],
+      );
+      if (JSON.parse(xhr.response).predictions[0]) {
+        node.classList.add('processed-true');
+        const tweetText = node.querySelector(
+          `${selector} > div ~ div > div ~ div`,
+        );
+        styleTweet(tweetText, option);
+      } else {
+        node.classList.add('processed-false');
+      }
+    } else {
+      console.error(e);
+      console.log('Failed response', xhr);
+    }
+  };
+  xhr.send(
+    JSON.stringify({
+      texts: [node.innerText],
+    }),
+  );
 };
 
 /*
-* Predefines action and changes it depending on user action
+ * Predefines action and changes it depending on user action
  */
+// disable eslint error for browser
```

Comment:
It seems that eslint does not know `browser` because apparently it comes from the context of the extension. For now I just ignored it, but maybe we can teach eslint to know it.